lib/models/event.dart

```diff
@@ -62,6 +62,21 @@ class Event extends Undertaking {
             timeCreated: e.timeCreated,
             timeModified: e.timeModified);
 
+  /// same thing as clone, except generate a new ID
+  Event.copy(Event e)
+      : this(
+            name: e.name,
+            id: "${e.id}-copy-${DateTime.now().microsecondsSinceEpoch}",
+            description: e.description,
+            color: e.color,
+            location: e.location,
+            tags: e.tags,
+            recurrenceRules: e.recurrenceRules,
+            timeStart: e.timeStart,
+            timeEnd: e.timeEnd,
+            timeCreated: e.timeCreated,
+            timeModified: e.timeModified);
+
```

Comment:
since the code looks mostly the same, could we change the clone function to have an optional parameter for generateNewId?

Reply:
we can give it a default value of true